pycloudlib/cloud.py

```diff
@@ -251,7 +251,12 @@ def use_key(self, public_key_path, private_key_path=None, name=None):
             name: name to reference key by
         """
         self._log.debug("using SSH key from %s", public_key_path)
-        self.key_pair = KeyPair(public_key_path, private_key_path, name)
+        # overwrite all 3 ssh key settings/values in config, even if any new values are None
```

Comment:
Is there a reason this section needed to change? I don't like the idea of modifying the config as it's supposed to represent the on disk configuration, not something that gets changed in place.

If the concern is something like `use_key()` can pass public key while `private_key_path` is sourced from the config, I think we can just say "not supported, don't do that".

Reply:
Yeah I wasn't sure if this made sense to do since normally the config is not mutated. So I will revert this. I will have to refactor get_ssh_keys() or something though, since it relies on values being set in the config. That should be trivial to refactor though.